src/main/java/logging/LogParser.java

```diff
@@ -0,0 +1,52 @@
+package logging;
+
+import logging.enums.LogLevel;
+import logging.exception.InvalidJsonException;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class LogParser {
+
+    public String parseStringValueFromJsonStringByKey(String jsonString, String key) {
+        String result;
+        JSONObject jsonObject = tryParse(jsonString);
+        result = getStringValue(jsonObject, key);
+        return result;
+    }
+
+    private String getStringValue(JSONObject jsonString, String key) {
+        String result = null;
+        try {
+            result = jsonString.getString(key);
+        } catch (JSONException e) {
+            System.out.println("invalid JsonKey");
+        }
+        return result;
+    }
+
+    private JSONObject tryParse(String outPutFileNameInJson) {
+        JSONObject result;
+        try {
+            result = new JSONObject(outPutFileNameInJson);
+        } catch (JSONException e) {
+            throw new InvalidJsonException();
+        }
+        return result;
+    }
+
+    public LogLevel parseLogLevelFromJson(String logLevelInJson) {
+        String logLevelString = parseStringValueFromJsonStringByKey(logLevelInJson, "logLevel");
+        return convertStringToLogLevel(logLevelString);
+    }
+
+    private LogLevel convertStringToLogLevel(String logLevelString) {
```

Comment:
Szerintem ezt valahogy enummal is meglehet csinálni valueOF (https://www.baeldung.com/java-string-to-enum), Talán a enum BO-tól el is várható ez.

Reply:
Átalakítottam. A valueOf-al az volt a baj, hogy nem lehet felülírni és csak akkor működik, ha a kérés CSUPA NAGYBETŰVEL van írva. Úgy oldottam meg végül, hogy az enum osztálynak van egy getLevel(String value) függvénye, ami uppercase-esít és utána meghívja a valueOf-ot.